### PR TITLE
Metrics could be retrieved from metrics-collector with given order ty…

### PR DESCRIPTION
**Pull request by qibobo**
…pe(asc or desc) by timestamp

[#147079897]

### CODE REVIEW

File: src/autoscaler/metricscollector/server/metric_handler.go [hunk 1]
```diff
@@ -69,12 +71,14 @@ func (h *MetricHandler) GetMetricHistories(w http.ResponseWriter, r *http.Reques
...
 	endParam := r.URL.Query()["end"]
+	orderParam := r.URL.Query()["order"]
 
-	h.logger.Debug("get-metric-histories", lager.Data{"appId": appId, "metrictype": metricType, "start": startParam, "end": endParam})
+	h.logger.Debug("get-metric-histories", lager.Data{"appId": appId, "metrictype": metricType, "start": startParam, "end": endParam, "order": orderParam})
 
 	var err error
 	start := int64(0)
 	end := int64(-1)
+	order := db.DESC
```
Comment (boyang9527) [inline src/autoscaler/metricscollector/server/metric_handler.go]: why default to descending order?

File: src/autoscaler/metricscollector/server/metric_handler_test.go [hunk 2]
```diff
@@ -232,55 +234,111 @@ var _ = Describe("MetricHandler", func() {
...
+			})
+
 		})
 
 		Context("when request query string is valid", func() {
-			Context("when there are both start and end time in query string", func() {
+			Context("when start,end and order are all in query string", func() {
 				BeforeEach(func() {
-					req, err = http.NewRequest(http.MethodGet, testUrlMetricHistories+"?start=123&end=567", nil)
+					req, err = http.NewRequest(http.MethodGet, testUrlMetricHistories+"?start=123&end=567&order=desc", nil)
```
Comment (boyang9527) [inline src/autoscaler/metricscollector/server/metric_handler_test.go]: if DESC is default, you need to test ASC

File: src/autoscaler/metricscollector/server/metric_handler.go (hunk 1 above)
Comment (qibobo) [inline src/autoscaler/metricscollector/server/metric_handler.go]: default should be asc.

**Comment by qibobo** [comment]
@boyang9527  updated

**Comment by qibobo** [comment]
@kanekoh @rohitsharma04 could you help review this pr? thanks.

**Comment by boyang9527** [comment]
LGTM

File: src/autoscaler/db/sqldb/instancemetrics_sqldb_test.go [hunk 3]
```diff
@@ -212,8 +215,11 @@ var _ = Describe("InstancemetricsSqldb", func() {
 			})
 		})
 
-		Context("when retriving all the metrics( start = 0, end = -1) ", func() {
-			It("removes duplicates and returns all the instance metrics of the app ordered by timestamp", func() {
+		Context("when retriving all the metrics( start = 0, end = -1, orderType = ASC) ", func() {
```
Comment (ghaih) [inline src/autoscaler/db/sqldb/instancemetrics_sqldb_test.go]: Typo retrieving. Present at other spots as well.

Comment (qibobo) [inline src/autoscaler/db/sqldb/instancemetrics_sqldb_test.go]: @ghaih fixed.

**Pull request merged**